### PR TITLE
Record player_ready event in teehistorian

### PR DESCRIPTION
**Pull request by Zwelf**
Allows reproducing teehistorian files without lookahead and without relying on PlayerNew messages.

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [x] Tested in combination with possibly related configuration options
  - tested with teehistorian deactivated
- [x] Written a unit test if it works standalone, system.c especially
- [x] Considered possible null pointers and out of bounds array indexing
- [x] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/server/gamecontroller.cpp [hunk 1]
```diff
@@ -396,6 +396,12 @@ bool IGameController::OnEntity(int Index, vec2 Pos, int Layer, int Flags, int Nu
 void IGameController::OnPlayerConnect(CPlayer *pPlayer)
 {
 	int ClientID = pPlayer->GetCID();
+
+	if(GameServer()->TeeHistorianActive())
+	{
+		GameServer()->TeeHistorian()->RecordPlayerReady(ClientID);
+	}
+
```
Comment (heinrich5991) [inline src/game/server/gamecontroller.cpp:404]: Could you put this into `CGameContext::OnClientEnter` instead? `IGameController` doesn't interact with teehistorian so far.

**Comment by heinrich5991** [comment]
Is your work of reproducing teehistorian public somewhere?

**Comment by heinrich5991** [comment]
Is this something you want to increase the minor teehistorian version over?

File: src/game/server/gamecontroller.cpp (hunk 1 above)
Comment (Zwelf) [reply src/game/server/gamecontroller.cpp]: Somehow didn't see the function in OnClientEnter. Done.

**Comment by Zwelf** [comment]
> Is this something you want to increase the minor teehistorian version over?

True, absolutely

> Is your work of reproducing teehistorian public somewhere?

No, not yet, but I would be comfortable publishing that repo.

**Comment by heinrich5991** [comment]
bors r+

Thanks for working on teehistorian. :)

**Pull request merged**